include/roboteam_ai/control/positionControl/pathPlanning/NumTreesPlanning.h

```diff
@@ -0,0 +1,68 @@
+//
+// Created by ratoone on 09-12-19.
+//
+
+#ifndef RTT_NUMTREESPLANNING_H
+#define RTT_NUMTREESPLANNING_H
+
+#include <list>
+#include <roboteam_utils/Vector2.h>
+#include <control/numTrees/PathPoint.h>
+#include <utilities/GameStateManager.hpp>
+#include <utilities/Constants.h>
+#include <queue>
+#include <control/ControlUtils.h>
+#include <interface/api/Output.h>
+#include <control/ControlUtils.h>
+#include "control/positionControl/CollisionDetector.h"
+
+namespace rtt::ai::control{
+
+/**
+ * Path planning algorithm. See method computePath for details.
+ */
+class NumTreesPlanning {
+private:
+    using PathPointer = std::shared_ptr<rtt::ai::control::PathPoint>;
```

Comment:
oh god. If there is any way you can make this a unique pointer or anything else...

Reply:
Why does this even have to be apointer

Reply:
I will open an issue with this and fix it after removing the whole NumTrees thing. Right now it's so deep entrenched in the old algorithm that this refactoring is not worth at this time